pkg/sloop/store/typed/eventcounttablegen.go

```diff
@@ -270,3 +270,71 @@ func (t *ResourceEventCountsTable) getLastMatchingKeyInPartition(txn badgerwrap.
 	}
 	return false, &EventCountKey{}, nil
 }
+
+//todo: add unit tests
+func (t *ResourceEventCountsTable) RangeReadPerPartition(txn badgerwrap.Txn, keyPrefix *EventCountKey,
+	valPredicateFn func(*ResourceEventCounts) bool, startTime time.Time, endTime time.Time) (map[EventCountKey]*ResourceEventCounts, RangeReadStats, error) {
+	resources := map[EventCountKey]*ResourceEventCounts{}
+
+	stats := RangeReadStats{}
+	before := time.Now()
+	partitionList, err := t.GetUniquePartitionList(txn)
```

Comment:
should we make a new function that gets unique partitions but takes in start and end time to do the filtering there?